postgresql/resource_postgresql_default_privileges.go

```diff
@@ -268,8 +268,12 @@ func readRoleDefaultPrivileges(txn *sql.Tx, d *schema.ResourceData) error {
 	}
 
 	privilegesSet := pgArrayToSet(privileges)
-	d.Set("privileges", privilegesSet)
-	d.SetId(generateDefaultPrivilegesID(d))
+	privilegesEqual := resourcePrivilegesEqual(privilegesSet, d)
+
+	if !privilegesEqual {
+		d.Set("privileges", privilegesSet)
+		d.SetId(generateDefaultPrivilegesID(d))
```

Comment:
Why the `SetId` is in the `if`? It's not the case for `postgresql_grant` and I think we want to set it in any cases.

Currently this works anyway because the ID is based on fields that recreate the resource if they change so it will never change here, but this is still totally unrelated to privileges

Reply:
you are right, this was probably just a slip.
changed this with [c35441d](https://github.com/cyrilgdn/terraform-provider-postgresql/pull/339/commits/c35441d539a2dc3b7e08bfa9ae5c0573cd53ec3f)